scripts/zones/VeLugannon_Palace/mobs/Detector.lua

```diff
@@ -10,25 +10,51 @@ entity.onMobSpawn = function(mob)
     mob:setLocalVar("petCount", 1)
 end
 
+local getMobToSpawn = function(detector)
+    local ID = require("scripts/zones/VeLugannon_Palace/IDs")
+    local detectorID = detector:getID()
+    local canSpawnSteamCleaner = GetServerVariable("[POP]SteamCleaner") < os.time() and utils.contains(detectorID, ID.mob.STEAM_CLEANER_DETECTORS)
```

Comment:
what happens if someone holds steam cleaner for 4 hours? Will steam cleaner be attempted to spawn again by another detector? I suppose an `and not steamCleaner:isSpawned()` after reorganizing it after `local steamCleaner = GetMobByID(ID.mob.STEAM_CLEANER)` would fix that.